recommender/static/js/src/recommender.js

```diff
@@ -1,852 +1,930 @@
-						}
-						$(divArrowDown).find('.recommender_vote_score').html(result['newVotes'].toString());
-					}
-                    else {
-                        alert(result['error']);
+    /**
+     * Upload the screenshot of resource before writing (adding/editing) the
+     * submitted resource to database.
+     * @param {element} formDiv The submission form for the resource.
+     * @param {file} file The file of screenshot.
+     * @param {string} writeType The string indicating we are going to add or edit resource.
+     * @param {dictionary} data The resource to be written.
+     */
+    function writeResourceWithScreenshot(formDiv, file, writeType, data) {
+        $.ajax({
+            type: 'POST',
+            url: uploadScreenshotUrl,
+            data: file,
+            contentType: false,
+            cache: false,
+            processData: false,
+            async: false,
+            complete: function(result) {
+                /**
+                 * File uploading error:
+                 * 1. The provided file is in wrong file type: accept files only in jpg, png, and gif.
+                 * 2. The filesystem (e.g., Amazon S3) is not properly set
+                 * 3. Size of uploaded file exceeds threshold
+                 */
+                for (var key in uploadFileError) {
+                    if (result.responseText.indexOf(uploadFileError[key]) == 0) {
+                        alert(uploadFileErrorText[uploadFileError[key]]);
+                        $("input[name='file']", formDiv).val('');
+                        if (writeType == writeDatabaseEnum.ADD) { enableAddSubmit(); }
+                        else if (writeType == writeDatabaseEnum.EDIT) { enableEditSubmit(); }
+                        return;
                     }
-				}
-			});
-		});
+                }
+                /* Writing the resource to database */
+                data['description'] = result.responseText;
+                if (writeType == writeDatabaseEnum.ADD) { addResource(data); }
+                else if (writeType == writeDatabaseEnum.EDIT) { editResource(data); }
+            },
+        });
+    }
+
+    /**
+     * Bind the event for adding a resource into the recommender.
+     */
+    function bindResourceAddEvent() {
+        /* Entering the page for adding resources */
+        $('.resourceAddButton', element).click(function() {
+            Logger.log('mit.recommender.addResource', generateLog(loggerStatus['addResource']['attempt']));
+        
+            resetAddResourcePage();
+            showModifyingPage('.addResourcePage');
+        });
+
+        /* If the input (text) area is changed, check whether user provides enough information for the resource */
+        $('.inTitle,.inUrl,.inDescriptionText', element).bind('input propertychange', function() { enableAddSubmit(); });
+        $('.addResourceForm', element).find("input[name='file']").change(function() {
+            if ($(this).val() != '') { enableAddSubmit(); }
```

Comment:
`!==`, unless you really want the automatic type coercion here. Which you almost certainly don't.